src/management/list-items.factory.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+/**
+ * @ngdoc service
+ * @name udb.management.listItems
+ * @description
+ * # Management list items
+ * Return the management list items to show in the sidebar as  a promise.
+ */
+angular
+  .module('udb.management')
+  .factory('managementListItems', listItems);
+
+/**
+ * @ngInject
+ * @return {Promise.<ManagementListItem[]>}
+ */
+function listItems(
+  RolePermission,
+  authorizationService,
+  ModerationManager,
+  $q,
+  managementListItemDefaults
+) {
+  var globalPermissionListItems = authorizationService
+    .getPermissions()
+    .then(generateListItems);
+
+  var moderationListItems = ModerationManager
+    .getMyRoles()
+    .then(generateModerationListItems);
+
+  return $q
+    .all([globalPermissionListItems, moderationListItems])
+    .then(_.flatten);
+
+  /**
+   * @param {Role[]} roles
+   * @return {number}
+   */
+  function countOffersWaitingForValidation(roles) {
+    var query = '';
+
+    _.forEach(roles, function(role) {
+      if (role.constraint) {
```

Comment:
@kln3wrld *role.constraint* is niet altijd gezet. Ik heb deze conditie er al ingestoken maar dit is niet voldoende. Je moet eigenlijk eerst filteren op rollen met een constraint. Als er dan nog rollen overblijven kan je pas de query opbouwen. Als je dat niet doet blijf je soms over met lege haakjes als query.